mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -175,7 +176,16 @@ impl UtxoRpcClientEnum {
                     },
                     Err(e) => {
                         if e.get_inner().is_tx_not_found_error() {
-                            return ERR!("Tx {} is not on chain anymore", tx_hash);
+                            error!(
+                                "Error {} getting tx from chain, retrying in 10 seconds. Retry left: {}",
+                                e, retry_req_count
+                            );
+                            Timer::sleep(check_every as f64).await;
+                            if retry_req_count == 0 {
+                                return ERR!("Tx {} is not on chain anymore", tx_hash);
+                            }
+                            retry_req_count -= 1;
```

Comment:
This check should be before the error logging and the `Timer::sleep` to return the error straight away when no retry attempts are left. Also I think the error msg here should be `Tx {} was not found on chain after {} tries error: {}`

Reply:
great idea!, thanks. done